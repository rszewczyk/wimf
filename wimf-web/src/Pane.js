// @flow
import React from "react";
import { css, merge } from "glamor";

type WhiteSpace = 1 | 2 | 3 | 4 | 5;

type PaneProps = {
  children?: any,
  border?: boolean,
  marginY?: WhiteSpace,
  marginX?: WhiteSpace
};

const paneStyle = css({
  padding: "1rem"
});

const withBorder = merge(paneStyle, {
  border: "1px solid gray",
  borderRadius: "3px"
});

const margins = ["0.5rem", "0.875rem", "1.0rem", "2.0rem", "4.0rem"];

export default function Pane(props: PaneProps) {
  const { border, marginX, marginY, children } = props;
  return (
    <div
      {...(border ? withBorder : paneStyle)}
      style={{
        marginTop: marginY ? margins[marginY] : 0,
        marginBottom: marginY ? margins[marginY] : 0,
        marginRight: marginX ? margins[marginX] : 0,
        marginLeft: marginX ? margins[marginX] : 0
      }}
    >
      {children}
    </div>
  );
}
