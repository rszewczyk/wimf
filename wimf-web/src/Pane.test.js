import React from "react";
import renderer from "react-test-renderer";
import Pane from "./Pane";

it("renders correctly", () => {
  const borderless = renderer
    .create(
      <Pane>
        <h1>hello!</h1>
        <p>foo</p>
      </Pane>
    )
    .toJSON();
  expect(borderless).toMatchSnapshot();

  const withBorder = renderer
    .create(
      <Pane border>
        <h1>hello!</h1>
        <p>foo</p>
      </Pane>
    )
    .toJSON();
  expect(withBorder).toMatchSnapshot();

  const withMargins = renderer
    .create(
      <Pane border marginX={1} marginY={2}>
        <h1>hello!</h1>
        <p>foo</p>
      </Pane>
    )
    .toJSON();
  expect(withMargins).toMatchSnapshot();
});
