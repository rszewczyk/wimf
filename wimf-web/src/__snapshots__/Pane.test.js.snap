// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  data-css-m0eymj=""
  style={
    Object {
      "marginBottom": 0,
      "marginLeft": 0,
      "marginRight": 0,
      "marginTop": 0,
    }
  }
>
  <h1>
    hello!
  </h1>
  <p>
    foo
  </p>
</div>
`;

exports[`renders correctly 2`] = `
<div
  data-css-emxoea=""
  style={
    Object {
      "marginBottom": 0,
      "marginLeft": 0,
      "marginRight": 0,
      "marginTop": 0,
    }
  }
>
  <h1>
    hello!
  </h1>
  <p>
    foo
  </p>
</div>
`;

exports[`renders correctly 3`] = `
<div
  data-css-emxoea=""
  style={
    Object {
      "marginBottom": "1.0rem",
      "marginLeft": "0.875rem",
      "marginRight": "0.875rem",
      "marginTop": "1.0rem",
    }
  }
>
  <h1>
    hello!
  </h1>
  <p>
    foo
  </p>
</div>
`;
